tensorboard/webapp/metrics/views/card_renderer/card_view_container.ts

```diff
@@ -62,6 +70,35 @@ export class CardViewContainer {
   @HostBinding('class.full-width') showFullWidth: boolean = false;
   @HostBinding('class.full-height') showFullHeight: boolean = false;
 
+  private readonly ngUnsubscribe = new Subject();
+  private readonly cardPinStateToggledSubject = new Subject();
+
+  ngOnInit() {
+    this.cardPinStateToggledSubject
+      .pipe(
+        takeUntil(this.ngUnsubscribe),
+        withLatestFrom(
+          this.store.select(selectors.getCardPinnedState, this.cardId),
+          this.store.select(selectors.getCanCreateNewPins)
+        ),
+        tap(([toggledAction, wasPinned, canCreateNewPins]) => {
+          this.store.dispatch(
+            actions.cardPinStateToggled({
+              cardId: this.cardId,
+              canCreateNewPins,
+              wasPinned: wasPinned,
+            })
+          );
+        })
+      )
+      .subscribe(() => {});
```

Comment:
alternatively, can we just use `this.store.select().pipe(take(1),...)`? down below in `onPinStateChanged`? Is this pattern better?